code/lib/preview-web/src/PreviewWeb.tsx

```diff
@@ -49,33 +51,45 @@ function focusInInput(event: Event) {
   return /input|textarea/i.test(target.tagName) || target.getAttribute('contenteditable') !== null;
 }
 
-type PossibleRender<TFramework extends AnyFramework> =
-  | StoryRender<TFramework>
-  | TemplateDocsRender<TFramework>
-  | StandaloneDocsRender<TFramework>;
+type PossibleRender<TFramework extends AnyFramework, TRootElement = HTMLElement> =
+  | StoryRender<TFramework, TRootElement>
+  | TemplateDocsRender<TFramework, TRootElement>
+  | StandaloneDocsRender<TFramework, TRootElement>;
 
-function isStoryRender<TFramework extends AnyFramework>(
-  r: PossibleRender<TFramework>
-): r is StoryRender<TFramework> {
+function isStoryRender<TFramework extends AnyFramework, TRootElement>(
+  r: PossibleRender<TFramework, TRootElement>
+): r is StoryRender<TFramework, TRootElement> {
   return r.type === 'story';
 }
 
-export class PreviewWeb<TFramework extends AnyFramework> extends Preview<TFramework> {
-  urlStore: UrlStore;
+export class PreviewWeb<
```

Comment:
Really we should probably rename this class to like `PreviewWithSelection` or something, as the web part is now parameterized. Might be something to do in the preview refactor @ndelangen.

So we have:
1. `Preview` -- something that can render a story to an element and listen to the channel to update it
2. `PreviewWeb` -- a sub class that also cares about the "current selection" and renders that to a special element in the view.

Reply:
I'm going to make these changes later when doing preview package refactor.